File: mobile/lib/domain/models/login_response.dart
/// doc/API_SPEC.md 3.1 — LoginResponse
/// 모바일(X-Client: mobile): refreshToken 본문 포함 (RULE 6.1.6)
class LoginResponse {
  const LoginResponse({
    required this.accessToken,
    required this.tokenType,
    required this.expiresIn,
    this.refreshToken,
  });

  final String accessToken;
  final String tokenType;
  final int expiresIn;
  final String? refreshToken;

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      accessToken: json['accessToken'] as String,
      tokenType: json['tokenType'] as String,
      expiresIn: (json['expiresIn'] as num).toInt(),
      refreshToken: json['refreshToken'] as String?,
    );
  }
}

File: src/main/java/com/example/sns/dto/response/LoginResponse.java
package com.example.sns.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;

/**
 * 로그인·토큰 갱신 응답 DTO.
 *
 * API 명세: accessToken, tokenType, expiresIn.
 * 모바일(X-Client: mobile): refreshToken 본문 포함 (RULE 6.1.6).
 */
public record LoginResponse(
        String accessToken,
        String tokenType,
        long expiresIn,
        @JsonInclude(JsonInclude.Include.NON_NULL) String refreshToken
) {

    public static LoginResponse of(String accessToken, long expiresInSeconds) {
        return new LoginResponse(accessToken, "Bearer", expiresInSeconds, null);
    }

    /** 모바일용: Refresh Token 본문 포함 (쿠키 미지원 환경) */
    public static LoginResponse of(String accessToken, long expiresInSeconds, String refreshToken) {
        return new LoginResponse(accessToken, "Bearer", expiresInSeconds, refreshToken);
    }
}
